### PR TITLE
IDF conversion support

### PR DESCRIPTION
**Pull request by Huite**

### CODE REVIEW

File: imodqgis/idf/conversion.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    def nband(self) -> int:
+        return self.__raster__.RasterCount
+
+
+def read_idf(path: str) -> Tuple[Dict[str, Any], np.ndarray]:
+    """Read the IDF header information into a dictionary"""
+    attrs = {}
+    with open(path, "rb") as f:
+        reclen_id = struct.unpack("i", f.read(4))[0]  # Lahey RecordLength Ident.
+        if reclen_id == 1271:
```
Comment (JoerivanEngelen) [inline imodqgis/idf/conversion.py:85]: You could store this stuff in a constant, e.g. RECLEN_SINGLE

Or maybe a dataclass with e.g. and instance ``SINGLE``

```python

from dataclasses import dataclass


@dataclass
class Precision:
    reclen: int
    floatsize: int
    intsize: int
    floatformat: str
    intformat: str
    dtype: str
    doubleprecision: bool

SINGLE = Precision(
    reclen=1271,
    floatsize=4,
    intsize=4,
    floatformat="f",
    intformat="i",
    dtype="float32",
    doubleprecision=False,
)

```

File: imodqgis/idf/idf_dialog.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        idf_path = self.line_edit.text()
+        gdal_path = layer.dataProvider().dataSourceUri()
+        convert_gdal_to_idf(gdal_path, idf_path, dtype)
+
+        self.parent.message_bar.pushMessage(
+            title="Info",
+            text=f"Exported {layer.name()} to {idf_path}",
+            level=Qgis.Info,
+        )
+        return None
```
Comment (JoerivanEngelen) [inline imodqgis/idf/idf_dialog.py:169]:
```suggestion
        return
```

**Review by JoerivanEngelen — COMMENTED** [review]
Looks good, I had a comment, but nothing critical.

File: imodqgis/idf/conversion.py (hunk 1 above)
Comment (Huite) [reply imodqgis/idf/conversion.py:85]: I think this is more hassle than it's worth, we just need the variables in the function body.

**Pull request merged**